### PR TITLE
#20192 md generation wasnt working on the db when fp has been written

### LINKED ISSUE #20154: Focal Point is lost when hitting save on the image editor and then publish.
**Describe the bug**
Focal point MD isn't set from content version to version 

**To Reproduce**
1. Let's start by uploading a brand new Image File-Asset
2. Click on Publish
3. Then Open-Up the image editor by double-clicking on the image. Set a Focal Point anywhere.
4. Hit Save
5. At the Content Editor level - Hit Publish.
6. Close and re-open the  Content Editor dialog and verify the Image doesn't have any Focal-point

**Expected behavior**
The Focal Point data must be persisted from version to version after hitting publish. And now it isn't.
Additionally to the compress drop-down isn't working on the Image Editor. And that also needs to be fixed.

### LINKED ISSUE #20192: File Metadata does not get generated if custom attributes are initially written.
**Description**
The real useful File Metadata doesn't get generated if the binary already has custom attributes
This is because the basic heuristic used to generate the metadata says. if it already has metadata do not generate it. 
And the finder method does not differentiate if the metadata present on the binary is made up exclusively from custom attributes. it only states that it already has metadata associated. Therefore the other metadata is never generated.
 
Additionally, when copying metadata from version to version we only must copy custom attributes. if it does make sense. 
I've seen cases in which a binary gets replaced from an image to pdf or txt. and stuff like that.

**To Reproduce**
1. Upload an image, then before publishing it. Open the Image Editor and add a focal point 
2. Hit Save and then Publish the close the window.
3. Now Re-open the Content editor for the same file and verify the MD
4. See There is no MD except for the focal point

If you initially publish the file and then come back to add the focal point it'll work as expected.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This basically fixes a problem generating  and storing the MD on the DB 
that had already been fixed for the File System implementation.  

The problem was that if the MD had been previously generated only with custom attributes the MD generator would skip the generation of the file MD attributes because it would think there was already MD for that binary.

Additionally to that. Another Erratic behavior on the API was corrected. Where a method that would force the MD generation was getting called from within the Content. The rule needs to be.. We Only generate metadata when indexing. If such mechanism is disconnected the generation of the metadata occurs when calling the API directly or calling `Contentl.getBinaryMetadata()` Method. Calling the method `fileMetadataAPI.getMetadata() `will serve the MD only when it is ready. This behavior helps us put in place an API that behaves predictably and more consistently

### CODE REVIEW

**Comment by erickgonzalez** [comment]
Please check these tests, seems to be related to MD?? https://raw.githack.com/dotCMS/test-results/issue-20192-focal-point-prevents-md-gen/projects/core/current/integration/postgres/reports/html/classes/com.dotmarketing.portlets.contentlet.business.ContentletAPITest.html#Test_Copy_Metadata_On_CheckIn

File: dotCMS/src/integration-test/java/com/dotcms/storage/FileMetadataAPITest.java [hunk 1]
```diff
@@ -417,8 +424,8 @@ public void Test_Get_Metadata_No_Cache(final StorageType storageType) throws Exc
 
             Metadata fileAssetMeta = fileMetadataAPI
                     .getFullMetadataNoCache(fileAssetContent, FILE_ASSET);
-            //Expect metadata it's generation is forced in case it doesn't exist.
-            assertNotNull(fileAssetMeta);
+            //Expect No metadata it's generation until generateContentletMetadata is called
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/storage/FileMetadataAPITest.java:427]: it's "generated"?

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/storage/FileMetadataAPITest.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotcms/storage/FileMetadataAPITest.java]: corrected

**Comment by fabrizzio-dotCMS** [comment]
> Please check these tests, seems to be related to MD?? https://raw.githack.com/dotCMS/test-results/issue-20192-focal-point-prevents-md-gen/projects/core/current/integration/postgres/reports/html/classes/com.dotmarketing.portlets.contentlet.business.ContentletAPITest.html#Test_Copy_Metadata_On_CheckIn

Looks like this was caused by overlapping threads.

**Pull request merged**